### ISSUE TITLE
expect statement changes results of downstream expect statements

### ISSUE THREAD

**Issue by hp4k1h5** [vuejs/vue-test-utils#1031]
### Version
1.0.0-beta.25

### Reproduction link
[https://jsfiddle.net/chrisvfritz/50wL7mdz/#&togetherjs=rpijLU9whQ](https://jsfiddle.net/chrisvfritz/50wL7mdz/#&togetherjs=rpijLU9whQ)

### Steps to reproduce
>>>>this code passes with some warnings:

it('should sort tags by id asc and desc', () => {
      expect(wrapper.vm.sortType).toBe('sortByIdDesc')
      expect(wrapper.findAll(Tag).wrappers[0].props().tag).toBe(tags[1])
      wrapper.setData({ sortType: 'sortByIdAsc' })
      wrapper.vm.sortType = 'sortByIdAsc'
      expect(wrapper.vm.sortType).toBe('sortByIdAsc')
      expect(wrapper.vm.filteredTags[0]).toBe(tags[0])
      expect(wrapper.findAll(Tag).wrappers[0].props().tag).toBe(tags[0])
    })

>>>>this code fails on the last expect:

it('should sort tags by id asc and desc', () => {
      expect(wrapper.vm.sortType).toBe('sortByIdDesc')
      // expect(wrapper.findAll(Tag).wrappers[0].props().tag).toBe(tags[1])
      wrapper.setData({ sortType: 'sortByIdAsc' })
      wrapper.vm.sortType = 'sortByIdAsc'
      expect(wrapper.vm.sortType).toBe('sortByIdAsc')
      expect(wrapper.vm.filteredTags[0]).toBe(tags[0])
      expect(wrapper.findAll(Tag).wrappers[0].props().tag).toBe(tags[0])
    })

### What is expected?
one expect statement should not effect later expect statements, when there are no props set or changes made within the expect statement

### What is actually happening?
when you comment out the first findAll, the DOM does is not reloaded after changing the sortType, causing the second findAll to fail.
when you uncomment the first findAll, the DOM is reactive to prop changes and the the Tags are reloaded in the correct sort order 

---
though i have not provided a sandbox version, it should be obvious that running an expect statement with no side effects should not alter the behavior of later tests. 
in this case, just running that first findAll causes the dom to become reactive in a way that it is not prior to running the expect(findall)

**Comment by eddyerburgh** [comment]
Although I can't verify, because you don't have a reproduction. It sounds like this issue is caused by how Vue Test Utils implements synchronous updates. It will be fixed after Vue 2.5.18 is released, which includes synchronous update mode in core. 

A temporary workaround is to run your tests asynchronously:

```js
it("renders", () => {
  let wrapper = shallowMount(demo, { sync: false });
  wrapper.setData({selectedDate: testDate})
  setTimeout(() => {
    wrapper.vm.selectedDate.should.be.equal(testDate)
  done()  
  })
});
```

I'm going to close this issue since there isn't a reproduction, which means I can't verify that the issue is solved or not.

**Comment by hp4k1h5** [comment]
thanks for the confirmation. 
using async has improved the reliability of our tests. 
looking forward to the next iteration.
